nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -119,6 +119,30 @@ ${cfg.extraConfig}
 
   hasLocalPostgresDB = let args = cfg.database_args; in
     usePostgresql && (!(args ? host) || (elem args.host [ "localhost" "127.0.0.1" "::1" ]));
+
+  registerNewMatrixUser =
+    let
+      isIpv6 = x: lib.length (lib.splitString ":" x) > 1;
+      listener =
+        lib.findFirst (
+          listener: lib.any (
+            resource: lib.any (
+              name: name == "client"
+            ) resource.names
+          ) listener.resources
+        ) (lib.last cfg.listeners) cfg.listeners;
+    in
+    pkgs.writeShellScriptBin "register_new_matrix_user" ''
```

Comment:
I like the idea but I'm not sure it's a good idea to use the same name as the original script. This might cause confusion ,depending on the order, if both are in path I think.

Ideally we could use a pre or suffix that reflects the module. `matrix-synapse-register_new_matrix_user`.

But better ideas are welcome.